scripts/setup-llvm.py

```
@@ -179,6 +179,55 @@ def parse_version_tuple(text: str) -> tuple[int, ...]:
+#  sysroot.  System LLVM at /usr is outside that sysroot, so linking against it
+#  pulls in libraries built for the host's libc/libstdc++ which may be
+#  ABI-incompatible with the ones inside the sandbox.  Detect this and refuse to
+#  use any LLVM install that falls outside the compiler's sysroot.
+def compiler_sysroot() -> Path | None:
+    """Return the ``--sysroot`` from clang++'s per-target config, if any."""
+    compiler = shutil.which("clang++")
+    if not compiler:
+        return None
+
+    try:
+        triple = subprocess.check_output([compiler, "-dumpmachine"], text=True).strip()
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)